### ISSUE TITLE
Another false-positive related to interpolation in `Lint/FormatParameterMismatch`

### ISSUE THREAD

**Issue by movermeyer** [rubocop/rubocop#11521]
Related to https://github.com/rubocop/rubocop/pull/11464 and https://github.com/rubocop/rubocop/pull/11487, here's another false positive in `Lint/FormatParameterMismatch`.

--------

## Expected behavior

No `Lint/FormatParameterMismatch` offenses found.

## Actual behavior

```
rubocop lib/shopify-i18n/michael.rb
Inspecting 1 file
W

Offenses:

lib/shopify-i18n/michael.rb:5:8: W: Lint/FormatParameterMismatch: Number of arguments (1) to format doesn't match the number of fields (0).
Kernel.format("%.#{decimal_places}f", number)
       ^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

number = 123.456
decimal_places = 2
result = Kernel.format("%.#{decimal_places}f", number)
puts(result)

```

## RuboCop version

```
rubocop -V
1.44.1 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.2) [arm64-darwin21]
  - rubocop-minitest 0.27.0
  - rubocop-performance 1.15.2
  - rubocop-rake 0.6.0
```
